Mathlib/Tactic/Positivity/Basic.lean

```diff
@@ -460,6 +463,25 @@ def evalIntCast : PositivityExt where eval {u α} _zα _pα e := do
   | .none =>
     pure .none
 
+/-- Extension for Rat.cast. -/
+@[positivity Rat.cast _]
+def evalRatCast : PositivityExt where eval {u α} _zα _pα e := do
+  let (.app _ (a : Q(ℚ))) ← whnfR e | throwError "not Rat.cast"
```

Comment:
I wanted to make this a `~q` match, like this
```
def evalRatCast : PositivityExt where eval {u} (α : Q(Type u)) _zα _pα (e : Q($α)) := do
  let _rα ← synthInstanceQ (q(RatCast $α))
  let ~q(Rat.cast ($a : ℚ)) := e | throwError "failed to match Rat.cast"
...
```
but I couldn't figure out how to make it work.

Reply:
Qq needs explicit type annotations on most `let`s

Reply:
Thank you!